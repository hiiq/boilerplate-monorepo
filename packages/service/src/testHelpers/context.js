import { User, Utils } from '@boilerplate-monorepo/common';

const stubLoader = name => ({
  clear: td.func(`${name}.clear`),
  load: td.func(`${name}.load`),
});

const db = {
  user: {
    read: jest.fn().mockName('read'),
    readRaw: jest.fn().mockName('readRaw'),
    recoveryFind: jest.fn().mockName('recoveryFind'),
    register: jest.fn().mockName('register'),
    revokeRefreshTokens: jest.fn().mockName('revokeRefreshTokens'),
    save: jest.fn().mockName('save'),
  },
  users: {
    read: jest.fn().mockName('read'),
    readPaginated: jest.fn().mockName('readPaginated'),
    readRaw: jest.fn().mockName('readRaw'),
  },
};

const make = Utils.makeSafeExample({
  db,
  ip: '127.0.0.1',
  loaders: {
    user: stubLoader('user'),
  },
  req: {
    headers: {},
    ip: '127.0.0.1',
  },
  res: {
    cookie: jest.fn().mockName('cookie'),
  },
  user: User.apiExample(),
});

export { make };
