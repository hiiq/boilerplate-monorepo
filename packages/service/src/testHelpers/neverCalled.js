const neverCalled = { ignoreExtraArgs: true, times: 0 };

export { neverCalled };
