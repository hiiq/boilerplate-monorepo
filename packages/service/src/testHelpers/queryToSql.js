import { pick } from 'ramda';

const queryToSql = pick(['bindings', 'sql', 'transaction']);

export { queryToSql };
