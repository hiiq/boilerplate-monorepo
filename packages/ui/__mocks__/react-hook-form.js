import React from 'react';

export const useForm = () => ({
  errors: {},
  formState: {
    isSubmitting: false,
    isValid: true,
  },
  handleSubmit: () => jest.fn().mockName('handleSubmit(fn)'),
  register: jest.fn().mockName('register'),
  reset: jest.fn().mockName('reset'),
  watch: jest.fn().mockName('watch'),
});

export const useFormContext = useForm;

export const FormContext = (props) => {
  const formProps = useFormContext();

  return <x-FormContext {...props} {...formProps} />;
};
