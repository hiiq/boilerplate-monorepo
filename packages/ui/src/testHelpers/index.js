import { cleanup, render } from '@testing-library/react';
import { curry } from 'ramda';
import React from 'react';
import { Providers } from 'modules/app';

export * from '@testing-library/react';

const customRender = curry((Component, defaultProps = {}, overrides = {}) =>
  render(<Component {...defaultProps} {...overrides} />, {
    wrapper: Providers,
  })
);

export { cleanup, customRender as render };
