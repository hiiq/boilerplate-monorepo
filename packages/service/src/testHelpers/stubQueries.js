import knexMock from 'mock-knex';
import { queryToSql } from './queryToSql';

const stubQueries = (mutateFn, responses = []) => {
  const tracker = knexMock.getTracker();

  tracker.install();

  let queries = [];

  tracker.on('query', (query, step) => {
    queries = [...queries, queryToSql(query)];
    mutateFn(queries);
    query.response(responses[step - 1]);
  });
};

export { stubQueries };
