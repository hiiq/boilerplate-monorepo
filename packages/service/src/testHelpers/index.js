import { make as makeContext } from './context';
import {
  graphqlClient,
  makeGraphqlClient,
  responseToData,
  responseToErrorCode,
} from './graphql';
import { neverCalled } from './neverCalled';
import { queryToSql } from './queryToSql';
import { stubQueries } from './stubQueries';

export {
  graphqlClient,
  makeContext,
  makeGraphqlClient,
  neverCalled,
  queryToSql,
  responseToData,
  responseToErrorCode,
  stubQueries,
};
