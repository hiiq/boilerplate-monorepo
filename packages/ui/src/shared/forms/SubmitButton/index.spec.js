import React from 'react';
import { FormContext } from 'react-hook-form';
import { render } from 'testHelpers';
import { SubmitButton } from '.';

jest.mock('../../Button');

describe('<SubmitButton/>', () => {
  const ariaLabel = 'LABEL';
  const defaultProps = {
    onClick: jest.fn().mockName('onClick'),
    text: ariaLabel,
  };

  // eslint-disable-next-line react/prop-types
  const wrapper = ({ children }) => <FormContext>{children}</FormContext>;

  const renderComponent = overrides =>
    render(<SubmitButton {...defaultProps} {...overrides} />, { wrapper });

  test('renders a submit button', () => {
    const { container } = renderComponent();

    expect(container.firstChild).toMatchSnapshot();
  });
});
