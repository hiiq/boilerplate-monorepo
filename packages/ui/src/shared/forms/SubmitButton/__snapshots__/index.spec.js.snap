// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitButton/> renders a submit button 1`] = `
<x-formcontext
  errors="[object Object]"
  formstate="[object Object]"
>
  <x-button
    context="PRIMARY"
    disabled="false"
    text="LABEL"
    type="submit"
  />
</x-formcontext>
`;
